azext_iot/iothub/providers/discovery.py

```diff
@@ -4,230 +4,98 @@
 # Licensed under the MIT License. See License.txt in the project root for license information.
 # --------------------------------------------------------------------------------------------
 
-from knack.util import CLIError
 from knack.log import get_logger
 from azure.cli.core.commands.client_factory import get_subscription_id
+from azext_iot.common._azure import CONN_STR_TEMPLATE
```

Comment:
We need a more specific name for CONN_STR_TEMPLATE being imported from `azext_iot.common._azure` because azure resources don't all share the same connection string form. The DPS cstring template can point to the same value if its the same form.